components/notifier/src/models/metric_notification_data.py

```diff
@@ -4,7 +4,7 @@
 class MetricNotificationData:  # pylint: disable=too-few-public-methods
     """Handle metric data needed for notifications."""
 
-    def __init__(self, metric, data_model):
+    def __init__(self, metric, data_model, reason):
```

Comment:
Reason is a string I assume?
```suggestion
    def __init__(self, metric, data_model, reason: str) -> None:
```